token.go

```diff
@@ -192,12 +192,14 @@ func (t *tokenRing) String() string {
 	return string(buf.Bytes())
 }
 
-func (t *tokenRing) GetHostForPartitionKey(partitionKey []byte) (host *HostInfo, endToken token) {
+func (t *tokenRing) GetHostForPartitionKey(partitionKey []byte) (host *HostInfo, endToken, actualToken token) {
 	if t == nil {
```

Comment:
Do we ever need endToken? If we return the host, why do we care about endToken? Appears to be only used for testing.

Reply:
No as far as I can tell we don't need it. Being conservative mostly I envisoned this going into the upstream driver as well. Silently exchanging the token of the current query with another seems confusing at best. The partitioner is available both on the token ring and at the current callers site so we could separate them both completely and simply pass the token to the `GetHostFor*` methods.

Reply:
Actually no the partitioner is not available in the policy. I think these methods should return the actual tokens. If we ever need the token boundaries we can create a new method that does just that explicitly.

Reply:
Another way to structure this is to add functions

    func (t* tokenRing) GetTokenForPartitionKey(partitionKey byte[]) token
    func (t* tokenRing) GetHostForToken(t token) (host *hostInfo, endToken token)

this separates the process of extracting the token and finding the host for it. The original function can call the two new functions in sequence.

Reply:
I can never learn Github commenting. This was resolved and then it silently doesn't show your answer. That would be a good change @avikivity  let me see if I can rebase it into the one that got merged.

Reply:
I added that comment only later.